operator/internal/controller/auto/update_controller.go

```diff
@@ -332,6 +334,11 @@ func (u *reconcileSession) Preview(ctx context.Context, obj *autov1alpha1.Update
 	reader := streamReader[agentpb.PreviewStream]{receiver: res, l: l, u: u, obj: obj}
 	_, err = reader.Result()
 	if err != nil {
+		// Update the status/conditions.
+		if err := u.updateStatus(ctx, obj); err != nil {
```

Comment:
Previously, when we get a grpc error, we do not surface this back to the Update CR's status/conditions. This ensures we always update the conditions/status.